contrib/mod_sftp/packet.c

```diff
@@ -127,6 +127,9 @@ static int packet_poll(int sockfd, int io) {
     "using select(2)", (unsigned long) tv.tv_sec, sockfd,
     io == SFTP_PACKET_IO_RD ? "reading" : "writing");
 
+  /* Clear any possibly stale pointers. */
+  session.curr_cmd_rec = NULL;
```

Comment:
This change is not necessarily related to the SSH data buffering fixes.  It does, however, quell a stacktrace encountered by folks testing this change.  The stacktrace occurs when logging-related code uses this variable, and it points to a stale `cmd_rec`, one allocated from a pool that has already been destroyed.  Thus we want to clear it here, before reading any network packet, as a best-effort mitigation/cleanup of such stale pointers.